app/models/manageiq/providers/inventory/persister/builder.rb

```diff
@@ -228,7 +229,16 @@ def auto_model_class
         # return the base class.
         #
         # safe_constantize can return different similar class ( some Rails auto-magic :/ )
```

Comment:
We should change this...we know what the magic does...
```suggestion
        # safe_constantize can return different similar class if it is able to resolve the class in the hierarchy even though it isn't at the same hierarchy depth we are expecting.
```

That being said, if we change line 225 to full qualify with `::#{manager_class}` do we still end up with the same problem?

Reply:
> That being said, if we change line 225 to full qualify with ::#{manager_class} do we still end up with the same problem?

Yes, I checked the code for constantize and it explicitly drops leading `::` as it is assumed to be looking in the root namespace

https://github.com/rails/rails/blob/6-0-stable/activesupport/lib/active_support/inflector/methods.rb#L258-L260
https://github.com/rails/rails/blob/6-0-stable/activesupport/lib/active_support/inflector/methods.rb#L277-L278